feat(collections): create native collection view

cyndichin opened the pull request:
## Summary
Create native collection view (focused on UI portion)

## References 
IN-1549

## Implementation Details
* Created native collection view in UIKit using `CollectionViewController`, `CollectionViewModel` (based on `SlateDetailViewController`, `SlateDetailViewModel`)
* For the metadata of collection, created `CollectionMetadataCell` and `CollectionMetadataPresenter` (based on ArticleMetadataPresenter)
* For the stories, created `CollectionStoryViewModel` (based on `HomeRecommendationCellViewModel`) and uses `RecommendationCell`
* Added feature flag for native collections `perm.ios.native_collections`
* Added logic to open a collection on Saves in `SavedItemsListViewModel` as well as API service `CollectionService` to retrieve graphQL data which will be refactored in the future

## Test Steps
1. Navigate to a collection in Saves list
2. Tap on the collection
3. Confirm that the native collection appears in the proper layout and format

## PR Checklist:
- N/A Added Unit / UI tests - No UI at this point of time, unit tests will be worked on in future tickets
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

## Screenshots
https://github.com/Pocket/pocket-ios/assets/6743397/6a5f240a-514a-4810-ab4a-7c90d4629228

cyndichin commented:
For this PR, the focus was on the UI portion of native collections. Some work was added to load graphql data, but this code may change. Overall, how the stories are opened (remote / local) as well as actions for collections / stories will be addressed in future tickets.

Gio2018 commented on PocketKit/Sources/PocketKit/Collections/CollectionMetadataCell.swift:
```diff
@@ -0,0 +1,80 @@
+    private let titleTextView = ArticleComponentTextView()
+    private let introTextView = ArticleComponentTextView()
+
+    private var metaStack: UIStackView = {
```
nit: since it's a `UIStackView`, maybe we can tweak the property name to `metaStackView` for clarity? Same for the others

Gio2018 commented on PocketKit/Sources/PocketKit/Collections/CollectionMetadataPresenter.swift:
```diff
@@ -0,0 +1,111 @@
+        return mutable
+    }
+
+    func size(for availableItemWidth: CGFloat) -> CGSize {
```
maybe we can add a description to this method to explain what it does?

Gio2018 commented on PocketKit/Sources/PocketKit/Collections/CollectionMetadataPresenter.swift:
```diff
@@ -0,0 +1,111 @@
+        )
+    }
+
+    private static func height(of attributedString: NSAttributedString, width: CGFloat) -> CGFloat {
```
same here about the description

Gio2018 reviewed (approved):
Great work @cyndichin ! Like we discussed on Slack, we'll need to revisit the fetch logic; other than that, I left some minor optional comments, but I think we can merge this for now and move forward with the project!

cyndichin commented:
thanks @Gio2018 for the thorough review! I addressed your comments here: [feat(collections): add comments and clean up](https://github.com/Pocket/pocket-ios/pull/802/commits/263e9df44b24209ef2259ab928ba85d3e3976562)
